pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -738,6 +754,276 @@ var _ = Describe("With a running MachineSync Reconciler", func() {
+			}), timeout).Should(Succeed())
+
+			// Wait until the binding shows the patched values
+			Eventually(k.Object(policyBinding), timeout).Should(
+				SatisfyAll(
+					HaveField("Spec.ParamRef.Namespace",
+						Equal(capiNamespace.GetName())),
+
+					HaveField("Spec.MatchResources.NamespaceSelector.MatchLabels",
+						HaveKeyWithValue("kubernetes.io/metadata.name",
+							mapiNamespace.GetName())),
+				),
+			)
+
+			By("Creating a throwaway MAPI machine")
+			testMachine := mapiMachineBuilder.WithGenerateName("test-machine").Build()
+			Expect(k8sClient.Create(ctx, testMachine)).Should(Succeed())
+
+			By("Setting the throwaway MAPI machine AuthoritativeAPI to Cluster API")
+			Eventually(k.UpdateStatus(testMachine, func() {
+				testMachine.Status.AuthoritativeAPI = machinev1beta1.MachineAuthorityClusterAPI
+			})).Should(Succeed())
+
+			Eventually(k.Object(testMachine), timeout).Should(
+				HaveField("Status.AuthoritativeAPI", Equal(machinev1beta1.MachineAuthorityClusterAPI)))
+
+			Eventually(k.Update(testMachine, func() {
+				testMachine.ObjectMeta.Labels = map[string]string{"test-sentinel": "fubar"}
+			}), timeout).Should(MatchError(ContainSubstring("policy in place")))
+
+			By("Creating the CAPI infra machine")
+			Eventually(k8sClient.Create(ctx, capaMachine), timeout).Should(Succeed(), "capa machine should be able to be created")
+
+			By("Creating the MAPI machine")
+			mapiMachine = mapiMachineBuilder.WithName("test-machine").WithLabels(map[string]string{
+				"machine.openshift.io/cluster-api-cluster":      "ci-op-gs2k97d6-c9e33-2smph",
+				"machine.openshift.io/cluster-api-machine-role": "worker",
+				"machine.openshift.io/cluster-api-machine-type": "worker",
+				"machine.openshift.io/cluster-api-machineset":   "ci-op-gs2k97d6-c9e33-2smph-worker-us-west-2b",
+				"machine.openshift.io/instance-type":            "m6a.xlarge",
+			}).WithAnnotations(map[string]string{
+				"machine.openshift.io/instance-state": "running",
+			}).Build()
+			Expect(k8sClient.Create(ctx, mapiMachine)).Should(Succeed())
+
+			By("Creating the CAPI Machine")
+			capiMachine = capiMachineBuilder.WithName("test-machine").WithLabels(map[string]string{
+				"machine.openshift.io/cluster-api-cluster":      "ci-op-gs2k97d6-c9e33-2smph",
+				"machine.openshift.io/cluster-api-machine-role": "worker",
+				"machine.openshift.io/cluster-api-machine-type": "worker",
+				"machine.openshift.io/cluster-api-machineset":   "ci-op-gs2k97d6-c9e33-2smph-worker-us-west-2b",
+				"machine.openshift.io/instance-type":            "m6a.xlarge",
+				"cluster.x-k8s.io/cluster-name":                 "ci-op-gs2k97d6-c9e33-2smph",
+				"cluster.x-k8s.io/set-name":                     "ci-op-gs2k97d6-c9e33-2smph-worker-us-west-2b",
+				"node-role.kubernetes.io/worker":                "",
+			}).WithAnnotations(map[string]string{
+				"machine.openshift.io/instance-state": "running",
+			}).Build()
+			Expect(k8sClient.Create(ctx, capiMachine)).Should(Succeed())
```

Comment:
Should we use Eventually?

Reply:
https://github.com/openshift/cluster-capi-operator/pull/321/files#r2257610280